cli/azd/pkg/azsdk/storage/storage_file_share_service.go

```diff
@@ -0,0 +1,81 @@
+package storage
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azfile/service"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azfile/share"
+	"github.com/azure/azure-dev/cli/azd/pkg/account"
+)
+
+type FileShareService interface {
+	// Upload files from source path to a file share
+	UploadFiles(ctx context.Context, subId, fileShareUrl, source, dest string) error
```

Comment:
nit: single file? `UploadFile`?

Reply:
uploads all the files from the source (can be a folder)

Reply:
Could we name this `UploadPath` then or something?  I to was confused looking at the implementation and thinking that it only supported a single file, especially since all of the recursive logic is hidden away inside the client we are wrapping.